packages/flutter/lib/src/material/slider.dart

```diff
@@ -424,11 +413,30 @@ class _RenderSlider extends RenderBox implements SemanticsActionHandler {
   TextDirection _textDirection;
   set textDirection(TextDirection value) {
     assert(value != null);
-    if (_textDirection == value)
+    if (value == _textDirection)
       return;
     _textDirection = value;
-    // TODO(abarth): Update _labelPainter's text direction.
-    markNeedsPaint();
+    _updateLabelPainter(labelChanged: false);
+  }
+
+  void _updateLabelPainter({ @required bool labelChanged }) {
+    if (label != null) {
+      // TODO(abarth): Handle textScaleFactor. https://github.com/flutter/flutter/issues/5938
+      _labelPainter
+        ..text = new TextSpan(
+          style: _textTheme.body1.copyWith(fontSize: 10.0),
+          text: label,
+        )
+        ..textDirection = textDirection
+        ..layout();
+    } else {
+      _labelPainter.text = null;
+    }
+    if (labelChanged) {
+      markNeedsLayout();
+    } else {
+      markNeedsPaint();
```

Comment:
Don't you need a layout if the textDirection changes?  The shaping could work out differently.

Reply:
Ooh, good point.